src/main/thrift/parquet.thrift

```diff
@@ -914,6 +1040,32 @@ struct RowGroup {
   7: optional i16 ordinal
 }
 
+struct RowGroupV3 {
+  /** REMOVED from v1: columns.
+   * Instead, decode each FileColumnMetadataV3 individually as needed.
+   */
+
+  /** Total byte size of all the uncompressed column data in this row group **/
+  1: required i64 total_byte_size
+
+  /** Number of rows in this row group **/
+  2: required i64 num_rows
+
+  /** If set, specifies a sort ordering of the rows in this row group. */
+  3: optional list<SortingColumn> sorting_columns
```

Comment:
I wonder if this should be per file.  I guess the downside is expense concatenating files with different sort orders.

Reply:
I have no idea. Not knowing the precise use cases for this, I would err on the side of safety and not change this.

Reply:
I was going to make the same comment. I would imagine the point of `sorting_columns` is to tip off a query engine as to which columns can be used for filtering (and perhaps also columns used as a compount key?). I can't see how it would make sense for this to vary per row group. Since we're blowing things up, I'd vote to move this to FileMetaData.

Reply:
I think this is kind of the same rationale as the codecs (easily being able to concatenate row groups with different sort orders)

Reply:
That is fair, but why lose the information per row-group it seems some engines could make sure of this, although it is probably superceded by page indexes?